test/test_qaoa_construction.py

```diff
@@ -0,0 +1,97 @@
+"""Test the construction of the QAOA ansatz."""
+
+
+from unittest import TestCase
+
+from qiskit import QuantumCircuit, transpile
+from qiskit.circuit import Parameter
+from qiskit.circuit.library import QAOAAnsatz
+from qiskit.primitives import StatevectorEstimator
+from qiskit.quantum_info import SparsePauliOp
+from qiskit.transpiler import PassManager
+from qiskit.transpiler.passes.routing.commuting_2q_gate_routing import SwapStrategy
+
+from qopt_best_practices.transpilation.qaoa_construction_pass import QAOAConstructionPass
+from qopt_best_practices.transpilation.preset_qaoa_passmanager import qaoa_swap_strategy
+
+
+class TestQAOAConstruction(TestCase):
+    """Test the construction of the QAOA ansatz."""
+
+    def setUp(self):
+        """Set up re-used variables."""
+        self.estimator = StatevectorEstimator()
+
+        gamma = Parameter("γ")
+        cost_op = QuantumCircuit(4)
+        cost_op.rzz(2 * gamma, 0, 1)
+        cost_op.rzz(2 * gamma, 2, 3)
+        cost_op.swap(0, 1)
+        cost_op.swap(2, 3)
+        cost_op.rzz(2 * gamma, 1, 2)
+
+        self.cost_op_circ = transpile(cost_op, basis_gates=["sx", "cx", "x", "rz"])
+
+        self.cost_op = SparsePauliOp.from_list([("IIZZ", 1), ("ZZII", 1), ("ZIIZ", 1)])
+
+        self.config = {
+            "swap_strategy": SwapStrategy.from_line(list(range(4))),
+            "edge_coloring": {(idx, idx + 1): (idx + 1) % 2 for idx in range(4)},
+        }
+
+    def test_depth_one(self):
+        """Compare the pass with the SWAPs and ensure the measurements are ordered properly."""
+        qaoa_pm = qaoa_swap_strategy(self.config)
```

Comment:
In case you apply suggestion above:

```suggestion
        qaoa_pm = qaoa_swap_strategy_pm(self.config)
```